client/task-list/tasks/payments/PaymentGatewaySuggestions/index.js

```diff
@@ -0,0 +1,202 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { getHistory, getNewPath } from '@woocommerce/navigation';
+import {
+	OPTIONS_STORE_NAME,
+	ONBOARDING_STORE_NAME,
+	PAYMENT_GATEWAYS_STORE_NAME,
+} from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+import { useMemo, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { List, Placeholder as ListPlaceholder } from './components/List';
+import { Setup, Placeholder as SetupPlaceholder } from './components/Setup';
+import { WCPayMethodCard } from '../components/WCPayMethodCard';
+import './plugins/Bacs';
+
+const RECOMMENDED_GATEWAY_KEYS = [
+	'woocommerce_payments',
+	'mercadopago',
+	'stripe',
+];
+
+export const PaymentGatewaySuggestions = ( { query } ) => {
+	const { updatePaymentGateway } = useDispatch( PAYMENT_GATEWAYS_STORE_NAME );
+	const {
+		additionalSuggestions,
+		enabledSuggestions,
+		getPaymentGateway,
+		paymentGateways,
+		suggestions,
+		isResolving,
+		wcPaySuggestion,
+	} = useSelect( ( select ) => {
+		const gateways = select( PAYMENT_GATEWAYS_STORE_NAME )
+			.getPaymentGateways()
+			.reduce( ( map, gateway ) => {
+				map[ gateway.id ] = gateway;
+				return map;
+			}, {} );
+
+		const enabled = new Map();
+		const additional = new Map();
+		let wcPay = null;
+		const mappedSugggestions = select( ONBOARDING_STORE_NAME )
+			.getPaymentGatewaySuggestions()
+			.reduce( ( map, suggestion ) => {
+				const { id } = suggestion;
```

Comment:
We're calling the identifier `id` here now, although we refer to it as `key` in a lot of other places. Can we unify them all around `id`?

Reply:
That's the idea.  I updated the two you caught in `WCPayMethodCard` - are there others that I missed here?  Outside of the soon-to-be-removed local methods.

Reply:
Thanks @joshuatf , sorry I wasn't more specific here. I'm referring mainly to the other references to `key` in this file, such as `RECOMMENDED_GATEWAY_KEYS`, `paymentGatewayKey`, and `gatewayKey`.

Reply:
Awesome- thank you and updated!